go/protocol/chat1/extras.go

```diff
@@ -334,6 +335,52 @@ func MessageUnboxedDebugLines(ms []MessageUnboxed) string {
 	return strings.Join(MessageUnboxedDebugStrings(ms), "\n")
 }
 
+const (
+	VersionErrorMessageBoxed = "messageboxed"
+	VersionErrorHeader       = "header"
+	VersionErrorBody         = "body"
+)
+
+// NOTE: these values correspond to the maximum accepted values in
+// chat/boxer.go. If these values are changed, they must also be accepted
+// there.
+var MaxMessageBoxedVersion MessageBoxedVersion = MessageBoxedVersion_V3
+var MaxHeaderVersion HeaderPlaintextVersion = HeaderPlaintextVersion_V1
+var MaxBodyVersion BodyPlaintextVersion = BodyPlaintextVersion_V1
+
+// Check if this error has a version that is now able to be understood by our client.
+func (m MessageUnboxedError) ParseableVersion() bool {
```

Comment:
we can also store version/kind info on this error type so we can drop the parsing later, should we?

Reply:
Yes I think we should do that.